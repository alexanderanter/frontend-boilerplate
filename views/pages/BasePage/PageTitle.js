import styled from 'styled-components';

const PageTitle = styled.h1`
  text-align: center;
  text-transform: capitalize;
`;

export default PageTitle;
