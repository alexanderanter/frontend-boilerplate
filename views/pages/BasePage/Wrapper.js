import styled from 'styled-components';
import colors from '../../../constants/colors';

const Wrapper = styled.div`
  max-width: 1500px;
  background: ${colors.light};
  margin: 50px auto;
  padding: 50px;
`;

export default Wrapper;
