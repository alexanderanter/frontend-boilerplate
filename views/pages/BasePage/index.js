import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired
};

const BasePage = ({ children }) => <div>{children}</div>;

BasePage.propTypes = propTypes;

export default BasePage;
