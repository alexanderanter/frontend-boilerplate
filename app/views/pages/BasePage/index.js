import React from 'react';
import PropTypes from 'prop-types';
import Wrapper from './Wrapper';
import PageTitle from './PageTitle';

const defaultProps = {
  title: 'PageTitle'
};

const propTypes = {
  title: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired
};

const BasePage = ({ title, children }) => (
  <Wrapper>
    <PageTitle>{title}</PageTitle>
    {children}
  </Wrapper>
);

BasePage.defaultProps = defaultProps;
BasePage.propTypes = propTypes;

export default BasePage;
