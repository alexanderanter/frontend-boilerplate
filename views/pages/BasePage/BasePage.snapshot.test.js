import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import BasePage from './index';

describe('BasePage', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<BasePage {...props}>{props.children}</BasePage>);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {
      children: <p>Some Page</p>
    };
    wrapper = undefined;
  });

  describe('without props', () => {
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });
});
