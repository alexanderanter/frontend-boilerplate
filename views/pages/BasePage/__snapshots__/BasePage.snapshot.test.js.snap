// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePage without props Should match snapshot 1`] = `
<styled.div>
  <styled.h1>
    PageTitle
  </styled.h1>
  <p>
    Some Page
  </p>
</styled.div>
`;
